### PR TITLE
Fix chest minecart relooting mechanics

### LINKED ISSUE #8565: Chest minecarts are no longer affected by re-looting options.
### Expected behavior

Chest minecarts found in mine shafts are restocked with loot like all other loot chests

### Observed/Actual behavior

as of 1.19 chest minecarts are excluded from loot re-rolls

### Steps/models to reproduce

set up chest-relooting
locate chest minecart
loot minecart
allow specified time to pass
no loot

### Plugin and Datapack List

issue occurs in out of box setups, but here is my list

plugins
[19:21:12 INFO]: Plugins (32): AntiCreeper, AntiEndermanGrief, Chunky, CMI, CMILib, ConsoleSpamFix, CoreProtect, DiscordSRV, dynmap, Graves, InventoryRollbackPlus, LastLoginAPI, LibsDisguises, LuckPerms, MarriageMaster, OpenInv, OreAnnouncer, Plan, PlayerParticles, PlugManX (PlugMan), ProCosmetics, ProtocolLib, ShopGUIPlus, StaffPlusPlus, UltimateAutoRestart, Vault, VaultChatFormatter, voicechat, Votifier, VotingPlugin, WorldEdit, WorldGuard

datapack list
[19:21:16 INFO]: There are 17 data packs enabled: [vanilla (built-in)], [file/bukkit (world)], [file/armor statues v2.8.4.zip (world)], [file/custom nether portals v2.3.6.zip (world)], [file/double shulker shells v1.3.3.zip (world)], [file/fast leaf decay v2.0.6.zip (world)], [file/more mob heads v2.10.1.zip (world)], [file/nether portal coords v1.1.3.zip (world)], [file/player head drops v1.1.3.zip (world)], [file/silence mobs v1.1.3.zip (world)], [file/spawning spheres v1.1.3.zip (world)], [file/tag v2.1.3.zip (world)], [file/villager workstation highlights v1.1.3.zip (world)], [file/wandering trades hermit edition v1.7.0.zip (world)], [file/VanillaTweaks_c725925.zip (world)], [file/coordinates hud v1.2.3.zip (world)], [file/BlazeandCaves Advancements Pack 1.15.zip (world)]

### Paper version

paper version
[19:21:55 INFO]: Checking version, please wait...
[19:21:56 INFO]: This server is running Paper version git-Paper-265 (MC: 1.19.2) (Implementing API version 1.19.2-R0.1-SNAPSHOT) (Git: 117579c)
You are 5 version(s) behind
Download the new version at: https://papermc.io/downloads
Previous version: git-Paper-220 (MC: 1.19.2)

### Other

this seems to only affect chest minecarts, all other chests (as far as i know) are working as intended.

### PR DESCRIPTION
**Pull request by nullishamy**
Closes #8565.

### CODE REVIEW

**Comment by lynxplay** [comment]
Welcome to paper! :tada: 

Thanks for your first contribution, happy to see new faces :)
I sadly did not have the time yet to test your work, I hope to get that done this evening or tomorrow.
As this is merely an addition to a feature we add, it would fit nicely in the [original patch](https://github.com/PaperMC/Paper/blob/master/patches/server/0094-LootTable-API-Replenishable-Lootables-Feature.patch) that introduced this logic instead of its own new patch.

See https://github.com/PaperMC/Paper/blob/master/CONTRIBUTING.md#modifying-patches for an introduction on how to do that. If you have further questions in regard to this, feel free to ask in our [discord](https://discord.gg/papermc).

**Comment by nullishamy** [comment]
I did test it locally, so it should work, but of course I am not well versed in Paper testing.

**Comment by nullishamy** [comment]
Righto, that should be done now. I'd like to note that the guide doesn't tell you what tree should be modified (root, Paper-Server, Paper-API, etc). Would be better if it did (I can PR this too if that's alright #8593 ).

File: patches/server/0094-LootTable-API-Replenishable-Lootables-Feature.patch [hunk 1]
```diff
@@ -603,7 +603,25 @@ index e2a9782e0ae60eb5557dce0831084c5722687df6..9c15af9b622716ba8b21b8b1138ad1cd
              ContainerHelper.loadAllItems(nbt, this.getItemStacks());
          }
  
-@@ -184,4 +184,13 @@ public interface ContainerEntity extends Container, MenuProvider {
+@@ -104,13 +104,15 @@ public interface ContainerEntity extends Container, MenuProvider {
+ 
+     default void unpackChestVehicleLootTable(@Nullable Player player) {
+         MinecraftServer minecraftServer = this.getLevel().getServer();
+-        if (this.getLootTable() != null && minecraftServer != null) {
++        if (this.getLootableData().shouldReplenish(player) && this.getLootTable() != null && minecraftServer != null) { // Paper
```
Comment (Owen1212055) [inline patches/server/0094-LootTable-API-Replenishable-Lootables-Feature.patch]: ``this.getLootTable() != null`` check is not needed,
shouldReplenish does a null check already.

**Review by Owen1212055 — APPROVED** [review]
Functionality wise however, looks good.

**Comment by nullishamy** [comment]
Done.

**Review by lynxplay — APPROVED** [review]
Works great! Thanks for quickly incorporating owen's suggestion.

**Review by electronicboy — APPROVED** [review]

**Pull request merged**